### DISCUSSION TITLE
Including botan via cmake build give include errors

### DISCUSSION THREAD

**Discussion by hpacherkar** [randombit/botan#4343] [Q&A] [answered]
I am trying to include botan 3.5.0 in one of the cmake builds.
I added `find_package(Botan "3.5.0" REQUIRED)` to CMakeList which is able to resolve and add botan to the build.
During make I get 2 types of errors depending on how the botan header is used.
1. #include <botan/kyber.h>
![Screenshot from 2024-09-09 17-39-03](https://github.com/user-attachments/assets/b574cb7c-a8d3-4dea-8430-bf1f9ef2fea4)

2. #include<botan-3/botan/kyber.h>
![Screenshot from 2024-09-09 17-37-57](https://github.com/user-attachments/assets/434ccfb0-18b3-4fae-a27c-e65c732a7fd7)

How would I be able to include the headers properly if the standard way does not work and requires the folder prefix "botan-3" to be included as well, which in turn raises errors for included headers in that file.

**Comment by reneme** [comment]
Your first usage (`#include <botan/kyber.h>`) is the correct way and should find the headers. The "botan-3" prefix should be part of the include path passed to the compiler.

Could you share some more details on your environment? Did you build/install Botan from source or did you pull it from a package manager?

**Reply by hpacherkar** [reply]
I am using ubuntu 24.04.1, G++ v13.2 and CMake v3.28. I did build botan-3.5.0 (also tried some older version tags) from source and installed it using make install. I tried creating an example first compiling it with g++ as "g++ -std=c++20 -c kyber_lib.cpp -o kyber_lib.o -I/usr/local/include/botan-3 -L/usr/local/lib -lbotan-3 -fPIC"

**Reply by hpacherkar** [reply]
This is the part of cmakelist 
`
find_package(Botan 3.5.0 REQUIRED)
target_link_libraries(ue PRIVATE ${UE_SOURCES}
                              ${CMAKE_THREAD_LIBS_INIT}
                              ${Boost_LIBRARIES}
                              ${ATOMIC_LIBS}
                              ${CRYPTOPP_LIBRARIES}
                              Botan::Botan
                              )

get_target_property(BOTAN_INCLUDE_DIRS Botan::Botan INTERFACE_INCLUDE_DIRECTORIES)

message(STATUS  "Botan3 - ${BOTAN_INCLUDE_DIRS}")

target_include_directories(srsue PRIVATE ${CRYPTOPP_INCLUDE_DIRS}
                                         ${BOTAN_INCLUDE_DIRS})
 `
Message correctly prints out the path for include directory.

Still the 'make' says header not found with <botan/kyber.h>

**Reply by hpacherkar** [reply, marked as answer]
I have rewritten most of the CMakelist, most probably it happened due to nested code structure. Marking it answered for the time being. Hopefully won't encounter this again. Thank you.

**Reply by reneme** [reply]
Your usage of the `Botan::Botan` target looks correct to me. I'm taking your last comment that you found the problem to be on your end. Thanks for the update.

In general: We're shipping the cmake config only since the beginning of this year. So any feedback (or bug reports) on it are very welcome. I'm glad its working for you now.
